nntrainer/src/tensor_dim.cpp

```diff
@@ -44,11 +44,21 @@ void TensorDim::resetLen() {
   len = dim[0] * feature_len;
 }
 
+unsigned int TensorDim::getTensorDim(unsigned int idx) {
+  if (idx < 0 || idx > MAXDIM)
+    throw std::invalid_argument(
+      "[TensorDim] Tensor Dimension index should be between 0 and 4");
+
+  return dim[idx];
+}
+
 void TensorDim::setTensorDim(unsigned int idx, unsigned int value) {
-  if (value == 0) {
+  if (idx < 0 || idx > MAXDIM)
+    throw std::invalid_argument(
```

Comment:
how about `throw std::out_of_range` ? We might need to differentiate those two cases.

Reply:
done.